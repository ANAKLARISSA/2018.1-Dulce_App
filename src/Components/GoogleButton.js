import {googleButton as styles} from '../styles' ;
import React from 'react';
import {Text, Image} from 'react-native';
import {Container, Content, Button} from 'native-base';

class GoogleButton extends React.Component {

  render() {
    return (
      <Container>
        <Content>
          <Button
            rounded bordered iconLeft
            style={styles.container}
            onPress={() => this.props.onPress()}
          >
            <Image style = {styles.img} source={require('../../assets/img/google.png')} />
            <Text style = {styles.text}> Entrar com Google</Text>
          </Button>
        </Content>
      </Container>
    );
  }
}

export default GoogleButton;
