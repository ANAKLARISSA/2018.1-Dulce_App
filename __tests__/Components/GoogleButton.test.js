import React from 'react';
import GoogleButton from '../../src/Components/GoogleButton';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

require('bezier');

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer
    .create(<GoogleButton />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('testing onPressG ',() => {
  const onPress = jest.fn();
  const wrapper = shallow(<GoogleButton onPress={onPress}/>);
  const button = wrapper.find('Styled(Button)').at(0);
  button.simulate('press');
});