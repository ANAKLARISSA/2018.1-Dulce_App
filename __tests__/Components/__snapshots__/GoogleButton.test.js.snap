// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
    virtual={undefined}
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#FFF",
            "borderBottomWidth": 1,
            "borderColor": "#007aff",
            "borderLeftWidth": 1,
            "borderRadius": 57,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "borderWidth": 1,
            "elevation": null,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginBottom": 0,
            "marginTop": 15,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingHorizontal": 20,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
            "width": 280,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/img/google.png",
            }
          }
          style={
            Object {
              "height": 40,
              "resizeMode": "contain",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "fontFamily": "Raleway",
              "fontSize": 20,
            }
          }
        >
           Entrar com Google
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
